### ISSUE TITLE
[AC] [Crash] Wandering Nodes Assertion Failed

### ISSUE THREAD

**Issue by Day36512** [trickerer/Trinity-Bots#455]
Hey Trickerer. Getting a couple Assertion Failed messages every once and awhile. Attached are also crash logs.

Assertion message: #----------------------------------------------------------------------#
 
>> ASSERTION FAILED

# Location 'C:\AzerothCore\AzerothCore-wotlk-with-NPCBots\src\common\Debugging\Errors.h:81'
# Function 'ASSERT_NOTNULL_IMPL'
# Condition 'pointer'
# Message 'BotDataMgr::GetClosestWanderNode(me)'
 
#----------------

------------------------------------------------------#

[Crashes.zip](https://github.com/trickerer/Trinity-Bots/files/11450063/Crashes.zip)

**Comment by Day36512** [comment]
I'll see what I can do.

**Comment by trickerer** [comment]
Just enable debug logging, it will show which wandering bot spawns where
`Appender.NpcBots=1,5,0`
`Logger.npcbots=5,NpcBots Server`

**Comment by Day36512** [comment]
Here's a report from an individual who submitted to me. Had a few people turn on debug logg
[Trickerer.zip](https://github.com/trickerer/Trinity-Bots/files/11457784/Trickerer.zip)
ing:

**Comment by trickerer** [comment]
Not 100% if I deducted it right but https://github.com/trickerer/AzerothCore-wotlk-with-NPCBots/commit/9109b805c659a9a299432cb9486cd6cefd499bb7 is supposed to fix this one

**Comment by trickerer** [comment]
Please confirm if the issue is resolved.

**Comment by Day36512** [comment]
Hey Trickerer. looks like more crashes on most recent build related to wandering nodes:
[Crashes.zip](https://github.com/trickerer/Trinity-Bots/files/11493431/Crashes.zip)


>> ASSERTION FAILED

# Location 'C:\AzerothCore\AzerothCore-wotlk-with-NPCBots\src\server\game\AI\NpcBots\botdatamgr.cpp:2540'
# Function 'BotDataMgr::GetNextWanderNode'
# Condition '!links.empty()'

Turning logs back on and will have something better for you.

**Comment by Day36512** [comment]
[Crashes.zip](https://github.com/trickerer/Trinity-Bots/files/11493526/Crashes.zip)
Crash report with logs

**Comment by Day36512** [comment]
Including another log. Does seem the script is a catalyst  for it for one reason or another.
[Server.zip](https://github.com/trickerer/Trinity-Bots/files/11493595/Server.zip)

**Comment by trickerer** [comment]
This one in unreldated to original issue.

This crashlog says that:
Level 7 warrior bot Lyria spawned at "Trade District" WP 712 (1-10), went to "Stormwind City" WP 16 (1-10) and then couldn't find anywhere to go next.
So server crashed right after startup.

This could only happen if something silly was done to WPs in DB. Like removing all links from WP 16 and making one-way link 712->16 AND removing all other Alliance level 1-10 spawn points. Or (more likely) if something was changed in code to make this happen.

I made it so that most Alliance bots spawn in Elwynn. Bots traverse WP 16 like normal in all directions.

Sorry but I can't consider this a valid report.
